### ISSUE TITLE
Outputs on AWS::ServiceCatalog::CloudFormationProvisionedProduct causing errors

### ISSUE THREAD

**Issue by kylekluever** [aws-cloudformation/cfn-lint#3474]
### CloudFormation Lint Version

cfn-lint 1.5.0

### What operating system are you using?

Amazon Linux

### Describe the bug

Dynamic values under `Outputs` on `AWS::ServiceCatalog::CloudFormationProvisionedProduct` resource types cause errors.

I tried testing with the vanilla example template from AWS CloudFormation [docs page](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-servicecatalog-cloudformationprovisionedproduct.html#aws-resource-servicecatalog-cloudformationprovisionedproduct--examples) on `AWS::ServiceCatalog::CloudFormationProvisionedProduct`

Running this command:
```
pipenv run cfn-lint --template ./test/success/sc-product.yaml
```

Against the reproduction template gives this error
```
E1010 'Outputs.SCLambdaArn' is not one of ['RecordId', 'CloudformationStackArn', 'Outputs', 'ProvisionedProductId']
test/success/sc-product.yaml:17:41
```

### Expected behavior

!GetAtt on dynamic `Outputs` values should work and not cause E1010 errors.

### Reproduction template

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Description: Serverless Stack
Resources:
  SimpleLambda:
    Type: AWS::ServiceCatalog::CloudFormationProvisionedProduct
    Properties:
      ProductName: Basic Lambda
      ProvisioningArtifactName: '1.0'
            
  SimpleApiGateway:
    Type: AWS::ServiceCatalog::CloudFormationProvisionedProduct
    Properties:
      ProductName: API Gateway
      ProvisioningArtifactName: '1.0'
      ProvisioningParameters:
        - Key: LambdaArn
          Value: !GetAtt [SimpleLambda, Outputs.SCLambdaArn]
```

**Comment by kddejong** [comment]
Should have a fix in shortly.  We do this for CloudFormation nested stacks, custom resources, just need to enable this for `AWS::ServiceCatalog::CloudFormationProvisionedProduct` as well.

**Comment by kddejong** [comment]
I'm going to put in a workaround for this but need to figure out a better long term solution.

**Comment by kylekluever** [comment]
Thanks @kddejong !
